Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIWebNavigationEvent.h

```diff
@@ -54,6 +54,8 @@ class WebExtensionAPIWebNavigationEvent : public WebExtensionAPIObject, public J
     void removeListener(WebPage*, RefPtr<WebExtensionCallbackHandler>);
     bool hasListener(RefPtr<WebExtensionCallbackHandler>);
 
+    void setEventType(WebExtensionEventListenerType type) { m_type = type; }
```

Comment:
I think we can make this be part of the constructor if you add this constructor:

```
    explicit WebExtensionAPIWebNavigationEvent(ForMainWorld forMainWorld, WebExtensionAPIRuntimeBase& runtime, WebExtensionContextProxy& context, WebExtensionEventListenerType type)
        : WebExtensionAPIObject(forMainWorld, runtime, context)
    {
        m_type = type;
    }
```